JavaIntegration/src/main/java/org/openzen/zencode/java/module/converters/JavaAnnotatedType.java

```diff
@@ -0,0 +1,171 @@
+		}
+		if (element instanceof GenericArrayType) {
+			return of(ElementType.GENERIC_ARRAY, null, element);
+		}
+		if (element instanceof WildcardType) {
+			return of(ElementType.WILDCARD, null, element);
+		}
+
+		return null;
+	}
+
+	private static JavaAnnotatedType of(final AnnotatedElement element) {
+		if (element instanceof AnnotatedParameterizedType) {
+			return of(ElementType.ANNOTATED_PARAMETERIZED_TYPE, element, null);
+		}
+		if (element instanceof AnnotatedType) {
+			return of(ElementType.ANNOTATED_TYPE, element, null);
+		}
+
+		return null;
+	}
+
+	private static JavaAnnotatedType of(final ElementType elementType, final AnnotatedElement annotatedElement, final Type type) {
+		return new JavaAnnotatedType(elementType, annotatedElement, type);
+	}
+
+	private static RuntimeException invalid(final Object object) {
+		final String className = object == null ? null : object.getClass().getName();
+		return new IllegalArgumentException("Unable to convert " + object + " (" + className + ") to a JavaAnnotatedType");
+	}
+
+	public ElementType getElementType() {
+		return this.elementType;
+	}
+
+	public AnnotatedElement getAnnotatedElement() {
+		return this.annotatedElement;
+	}
+
+	public Type getType() {
+		return this.type;
+	}
+
+	@Override
+	public boolean isAnnotationPresent(final Class<? extends Annotation> annotationClass) {
+		return this.ifAnnotatedElement(() -> false, it -> it.isAnnotationPresent(annotationClass));
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public <T extends Annotation> T[] getAnnotationsByType(Class<T> annotationClass) {
+		return this.ifAnnotatedElement(() -> (T[]) Array.newInstance(annotationClass, 0), it -> it.getAnnotationsByType(annotationClass));
+	}
+
+	@Override
+	public <T extends Annotation> T getDeclaredAnnotation(Class<T> annotationClass) {
+		return this.ifAnnotatedElement(() -> null, it -> it.getDeclaredAnnotation(annotationClass));
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public <T extends Annotation> T[] getDeclaredAnnotationsByType(Class<T> annotationClass) {
+		return this.ifAnnotatedElement(() -> (T[]) Array.newInstance(annotationClass, 0), it -> it.getDeclaredAnnotationsByType(annotationClass));
+	}
+
+	@Override
+	public String getTypeName() {
+		return this.ifType(() -> "invalid type", Type::getTypeName);
+	}
+
+	@Override
+	public <T extends Annotation> T getAnnotation(Class<T> annotationClass) {
+		return this.ifAnnotatedElement(() -> null, it -> it.getAnnotation(annotationClass));
+	}
+
+	@Override
+	public Annotation[] getAnnotations() {
+		return this.ifAnnotatedElement(() -> new Annotation[0], AnnotatedElement::getAnnotations);
+	}
+
+	@Override
+	public Annotation[] getDeclaredAnnotations() {
+		return this.ifAnnotatedElement(() -> new Annotation[0], AnnotatedElement::getDeclaredAnnotations);
+	}
+
+	private <T> T ifAnnotatedElement(final Supplier<T> def, final Function<AnnotatedElement, T> block) {
```

Comment:
Suggestion
For parameter names, since `default` is a keyword, how about `fallback` (def could be default, definition, defender, ...)